libbeat/outputs/shipper/shipper.go

```diff
@@ -125,35 +147,42 @@ func (c *shipper) Connect() error {
 	if err != nil {
 		return fmt.Errorf("failed to fetch server information: %w", err)
 	}
-	c.serverID = serverInfo.GetUuid()
+	s.serverID = serverInfo.GetUuid()
 
-	c.log.Debugf("connection to %s (%s) established.", c.config.Server, c.serverID)
+	s.log.Debugf("connection to %s (%s) established.", s.config.Server, s.serverID)
+
+	go func() {
+		err := s.ackLoop(s.backgroundCtx)
+		if err != nil {
+			s.log.Errorf("acknowledgment loop stopped: %s", err)
+		}
+	}()
```

Comment:
On each `Connect()` we start a new `ackLoop`, the previous one is supposed to fail when the connection is interrupted, for the same reason `Publish` returned an error that caused `Connect()` to be called again. Most likely it will require some fine-tuning after we have the integration tests.